diff --git a/library/common.h b/library/common.h
index 3936ffdfe..5638b9fef 100644
--- a/library/common.h
+++ b/library/common.h
@@ -189,6 +189,7 @@ static inline void mbedtls_xor(unsigned char *r,
                                size_t n)
 {
     size_t i = 0;
+#if 0
 #if defined(MBEDTLS_EFFICIENT_UNALIGNED_ACCESS)
 #if defined(MBEDTLS_HAVE_NEON_INTRINSICS) && \
     (!(defined(MBEDTLS_COMPILER_IS_GCC) && MBEDTLS_GCC_VERSION < 70300))
@@ -230,6 +231,7 @@ static inline void mbedtls_xor(unsigned char *r,
     }
 #endif
 #endif
+#endif
 #endif
     for (; i < n; i++) {
         r[i] = a[i] ^ b[i];
